xbmc/video/VideoInfoScanner.cpp

```diff
@@ -402,6 +405,68 @@ namespace VIDEO
       m_database.SetPathHash(strDirectory, hash);
     }
 
+    // see if movie set artwork empty and could be updated locally from the movie set information folder
+    if (content == CONTENT_MOVIES &&
+        !items.IsEmpty() &&
+        !URIUtils::GetFileName(items[0]->GetPath()).empty() &&
+        !URIUtils::IsPlugin(items[0]->GetDynPath()))
+    {
+      CVideoInfoTag movieDetails;
+      if (!items.IsEmpty() && m_database.GetMovieInfo(items[0]->GetPath(), movieDetails))
+      {
+        if (!movieDetails.m_set.title.empty())
+        {
+          CVideoInfoTag setDetails;
+          m_database.GetSetInfo(movieDetails.m_set.id, setDetails);
+          std::map<std::string, std::string> setArt;
+          const bool isArt =
+              m_database.GetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, setArt);
+          if (!isArt || !setArt.count("poster"))
+          {
+            // No art (or no poster as that is used in library view)
+            // Use movie scraper to get original set title
+
+            // Current set title (may have been overridden in set.nfo)
+            std::string setTitle{setDetails.GetTitle()};
+            ScraperPtr scraper = m_database.GetScraperForPath(items[0]->GetPath());
+            CVideoInfoDownloader infoDownloader{scraper};
```

Comment:
To match the rest of the code style here:
`ScraperPtr scraper{m_database.GetScraperForPath(items[0]->GetPath())};
Maybe it could even be `const ScraperPtr`?